src/replay_syscall.cc

```diff
@@ -1269,6 +1269,7 @@ static void rep_process_syscall_arch(ReplayTask* t, ReplayTraceStep* step,
         }
       }
     }
+      /* FALLTHROUGH */
```

Comment:
Looks like rr does [have some C code](https://github.com/mozilla/rr/search?l=c) and GCC 7 raises warnings in C too. So just add `__attribute__((fallthrough))` as fallback on gcc 7 in additional to the LLVM one? (Unless the C code will never need a fallthrough switch....)